contrib/haproxy/Makefile

```diff
@@ -0,0 +1,187 @@
+
+# Sources
+SWAGGER_JSON ?= haproxy-dataplaneapi-1.2.json
+
+# The directory in which the client bindings are generated
+OUTPUT_DIR ?= ./openapi
+
+# These files will be removed from the OUTPUT_DIR after the client bindings
+# are generated.
+FILES_TO_DELETE_POST_GEN := go.mod go.sum .travis.yml git_push.sh
+
+## --------------------------------------
+## Help
+## --------------------------------------
+
+help: ## Display this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z0-9_-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+## --------------------------------------
+## Tooling Binaries
+## --------------------------------------
+
+TOOLING_BINARIES := $(GOLANGCI_LINT)
+tools: $(TOOLING_BINARIES) ## Build tooling binaries
+.PHONY: $(TOOLING_BINARIES)
+$(TOOLING_BINARIES):
+	make -C $(TOOLS_DIR) $(@F)
+
+## --------------------------------------
+## Build
+## --------------------------------------
+
+.PHONY: build
+build: ## Builds the HAProxy dataplane API client bindings
+	go build -v $(OUTPUT_DIR)
+
+## --------------------------------------
+## Generate
+## --------------------------------------
+
+.PHONY: generate-with-binary
+generate-with-binary: $(SWAGGER_JSON)
+generate-with-binary: ## Generates the HAProxy dataplane API client bindings (binary)
+	@$(MAKE) --no-print-directory verify-openapi-generator-binary
+	$(OPENAPI_GENERATOR_BIN) generate --input-spec $< --generator-name go --output $(OUTPUT_DIR)
+	@rm -fr $(addprefix $(OUTPUT_DIR)/,$(FILES_TO_DELETE_POST_GEN))
+
+.PHONY: generate-with-docker
+generate-with-docker: $(SWAGGER_JSON)
+generate-with-docker: ## Generates the HAProxy dataplane API client bindings (docker)
+	@mkdir -p $(OUTPUT_DIR)
+	docker run --rm \
+	  -v $(abspath $<):/openapi/swagger.json:ro \
+	  -v $(abspath $(OUTPUT_DIR)):/openapi/output \
+	  $(OPENAPI_GENERATOR_IMG) \
+	  generate --input-spec /openapi/swagger.json --generator-name go --output /openapi/output
+	@rm -fr $(addprefix $(OUTPUT_DIR)/,$(FILES_TO_DELETE_POST_GEN))
+
+.PHONY: generate
+ifneq (,$(strip $(OPENAPI_GENERATOR_BIN)))
+generate: generate-with-binary
+else
+generate: generate-with-docker
+endif
+generate:  ## Generates the HAProxy dataplane API client bindings (auto)
+	$(MAKE) --no-print-directory remove-dupe-opts
+	$(MAKE) --no-print-directory fix-backend-model
+	$(MAKE) --no-print-directory fix-frontend-model
+
+# Please see https://github.com/OpenAPITools/openapi-generator/issues/741
+# for more information on the issue of duplicate Opt structures.
+.PHONY: remove-dupe-opts
+remove-dupe-opts: | $(GOLANGCI_LINT)
+remove-dupe-opts: ## Removes duplicate Opt structs from the generated code
+	@for struct in $$(grep -h 'type .\{1,\} struct' $(OUTPUT_DIR)/*.go | grep Opts  | sort | uniq -c | grep -v '^      1' | awk '{print $$3}'); do \
+	  for f in $$(/bin/ls $(OUTPUT_DIR)/*.go); do \
```

Comment:
Hi @yastij,

We *could*, but since this is in its own `Makefile` I don't think it's strictly necessary for now. However, I agree that we could iterate on this in the future and create `contrib/haproxy/hack` for some of these functions.